recipe/meta.yaml

```diff
@@ -34,13 +32,68 @@ test:
     - udunits2 -h
     - udunits2 -H meter -W miles
 
+outputs:
+  - name: libudunits2
+    script: install.sh  # [unix]
+    script: install.bat  # [win]
+    files:
+      - lib/libudunits2.so.*  # [linux]
+      - lib/libudunits2.*.dylib  # [osx]
+      - share/udunits/*.xml  # [not win]
+      - Scripts/udunits2.dll  # [win]
+      - Library/share/udunits/*.xml  # [win]
+      - etc/conda/activate.d/udunits2-activate.bat  # [win]
+      - etc/conda/deactivate.d/udunits2-deactivate.bat  # [win]
+      - etc/conda/activate.d/udunits2-activate.sh  # Needed also for win
+      - etc/conda/deactivate.d/udunits2-deactivate.sh  # to support bash users
+      - etc/conda/activate.d/udunits2-activate.csh  # [not win]
+      - etc/conda/deactivate.d/udunits2-deactivate.csh  # [not win]
+      - etc/conda/activate.d/udunits2-activate.fish  # [not win]
+      - etc/conda/deactivate.d/udunits2-deactivate.fish  # [not win]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - cmake  # [win]
+        - make  # [not win]
+        - autoconf  # [not win]
+        - automake  # [not win]
+        - libtool  # [not win]
+        - texinfo  # [not win]
+      host:
+        - expat
+      run:
+      run_constrained:
+        - udunits2 {{ version }}.*
+
+  - name: udunits2
+    script: install.sh  # [unix]
+    script: install.bat  # [win]
+    build:
+      run_exports:
+        - {{ pin_subpackage('libudunits2') }}
```

Comment:
Jus for my education, could you explain why this is needed here but not in `libudunits2`?  I see that `expat` does things this way, too, so I trust that it's right.  I just don't really understand `run_exports` very well and would like to.

Reply:
I'll do my best ;) What `run_exports` does is saying "A package that has this package (`udunits2`) as a `build` or `host` requirement, should automatically get that package (`libudunits2`) as a `run` requirement.
That means that a package that depends on `udunits2` to build against the library will get an automatic `run` requirement that makes sure the dynamic library is available at run time.

On the other hand, it doesn't really make sense to have `libudunits2` as a `build` or `host` requirement because that's not enough to build against it.

Does that make sense?

Reply:
Got it, that makes perfect sense.